app/controllers/application_controller/advanced_search.rb

```diff
@@ -164,13 +164,21 @@ def adv_search_redraw_tree(tree)
       page << javascript_prologue
       tree_name = x_active_tree.to_s
       page.replace("#{tree_name}_div", :partial => "shared/tree", :locals => {:tree => tree, :name => tree_name})
+      # Redraw Advanced search for explorer screens (or storage, configuration scripts tree)
+      display_mode = params[:button] == 'save' ? params[:button] : nil
+      page.replace("adv_search_body",   :partial => "layouts/adv_search_body",   :locals => {:mode => display_mode})
+      page.replace("adv_search_footer", :partial => "layouts/adv_search_footer", :locals => {:mode => display_mode})
```

Comment:
@hstastna : sorry to be the shop stopper here, but: the method is called `adv_search_redraw_tree` so please, stick with the name.

Why not create a new method `adv_search_redraw_non_explorer` (as the comment suggests) and put the code there?

Reply:
When I tried to put the code anywhere else but adv_search_redraw_tree method, it did not work well. This is why it is there. Not very nice, but works well. Maybe I should have mentioned it in comment. But maybe I was doing something wrong. I will try to make it better, anyway. Thanks for the review! :)